### PR TITLE
Add default suffix

### LINKED ISSUE #3: Discussion: Infer suffix from base class name
If base class defines some interface, e.g. `Similarity`, the child classes often use this as suffix, e.g. `DotProductSimilarity` and  `CosineSimilarity`. In such case, using the base class as a default suffix may be useful.

Thus, I propose to extend `Resolver.from_subclasses` to use the (normalized) name of the base class as suffix, if None is provided. To disable this, a user could explicitly pass the empty string.

### PR DESCRIPTION
**Pull request by mberr**
Closes #3

### CODE REVIEW

File: src/class_resolver/api.py [hunk 1]
```diff
@@ -33,11 +33,13 @@ def __init__(
...
         :param base: The base class
         :param default: The default class
-        :param suffix: The optional shared suffix of all classes
+        :param suffix: The optional shared suffix of all classes. If None, use the base class' name for it.
         :param synonyms: The optional synonym dictionary
         """
         self.base = base
         self.default = default
+        if suffix is None:
+            suffix = normalize_string(base.__name__, suffix=None)
```
Comment (cthoyt) [inline src/class_resolver/api.py:42]: I think `suffix=None` is a bit redundant. Why include it?

File: src/class_resolver/api.py [hunk 2]
```diff
@@ -33,11 +33,13 @@ def __init__(
         :param classes: A list of classes
         :param base: The base class
         :param default: The default class
-        :param suffix: The optional shared suffix of all classes
+        :param suffix: The optional shared suffix of all classes. If None, use the base class' name for it.
```
Comment (cthoyt) [inline src/class_resolver/api.py:36]: Should mention that if you want alternative behavior (like no suffix) to give an empty string.

Comment (mberr) [reply src/class_resolver/api.py]: 09e0007

File: src/class_resolver/api.py (hunk 1 above)
Comment (mberr) [reply src/class_resolver/api.py]: ae25051

**Pull request merged**